spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -128,17 +107,16 @@ public MessageProperties toMessageProperties(AMQP.BasicProperties source, Envelo
 
 	private RabbitExtendedBindingProperties extendedBindingProperties = new RabbitExtendedBindingProperties();
 
-	public RabbitMessageChannelBinder(ConnectionFactory connectionFactory, RabbitProperties rabbitProperties) {
-		super(true, new String[0]);
+	RabbitExchangeQueueProvisioner provisioningProvider;
```

Comment:
No, I think we still have to remove those beans from the ctx anyway on unbind.
So, such a functionality in the Provisioner API would make sense, although I could live just with the Rabbit impl...